centreon/src/Core/Dashboard/Playlist/Application/UseCase/SharePlaylist/SharePlaylist.php

```diff
@@ -0,0 +1,225 @@
+
+namespace Core\Dashboard\Playlist\Application\UseCase\SharePlaylist;
+
+use Centreon\Domain\Contact\Interfaces\ContactInterface;
+use Centreon\Domain\Log\LoggerTrait;
+use Centreon\Domain\Repository\Interfaces\DataStorageEngineInterface;
+use Core\Application\Common\UseCase\ErrorResponse;
+use Core\Application\Common\UseCase\ForbiddenResponse;
+use Core\Application\Common\UseCase\InvalidArgumentResponse;
+use Core\Application\Common\UseCase\NoContentResponse;
+use Core\Application\Common\UseCase\NotFoundResponse;
+use Core\Contact\Application\Repository\ReadContactGroupRepositoryInterface;
+use Core\Contact\Application\Repository\ReadContactRepositoryInterface;
+use Core\Contact\Domain\Model\ContactGroup;
+use Core\Dashboard\Domain\Model\DashboardRights;
+use Core\Dashboard\Playlist\Application\Exception\PlaylistException;
+use Core\Dashboard\Playlist\Application\Repository\WritePlaylistShareRepositoryInterface;
+
+final class SharePlaylist
+{
+    use LoggerTrait;
+
+    /**
+     * @param DashboardRights $rights
+     * @param SharePlaylistValidator $validator
+     * @param ContactInterface $user
+     * @param WritePlaylistShareRepositoryInterface $shareRepository
+     * @param DataStorageEngineInterface $dataStorageEngine
+     * @param ReadContactGroupRepositoryInterface $contactGroupRepository
+     * @param ReadContactRepositoryInterface $contactRepository
+     */
+    public function __construct(
+        private readonly DashboardRights $rights,
+        private readonly SharePlaylistValidator $validator,
+        private readonly ContactInterface $user,
+        private readonly WritePlaylistShareRepositoryInterface $shareRepository,
+        private readonly DataStorageEngineInterface $dataStorageEngine,
+        private readonly ReadContactGroupRepositoryInterface $contactGroupRepository,
+        private readonly ReadContactRepositoryInterface $contactRepository,
+    ) {
+    }
+
+    /**
+     * @param int $playlistId
+     * @param SharePlaylistRequest $request
+     * @param SharePlaylistPresenterInterface $presenter
+     */
+    public function __invoke(
+        int $playlistId,
+        SharePlaylistRequest $request,
+        SharePlaylistPresenterInterface $presenter
+    ): void {
+        try {
+            if (! $this->rights->canCreate()) {
+                $this->error('Access Forbidden');
+                $presenter->presentResponse(new ForbiddenResponse(
+                    PlaylistException::accessNotAllowed()->getMessage()
+                ));
+
+                return;
+            }
+
+            $this->info('validating Playlist exists', ['playlist_id' => $playlistId]);
+            $this->validator->validatePlaylist($playlistId, $this->user, $this->rights->hasAdminRole());
+
+            $contactIds = array_map(fn(array $contact): int => $contact['id'], $request->contacts);
+            $contactGroupIds = array_map(fn(array $contactGroup): int => $contactGroup['id'], $request->contactGroups);
```

Comment:
```suggestion
            $contactIds = array_map(fn (array $contact): int => $contact['id'], $request->contacts);
            $contactGroupIds = array_map(fn (array $contactGroup): int => $contactGroup['id'], $request->contactGroups);
```
Not sure if it was carved that a space must be added after `fn`